capstone/src/main/java/com/google/univiz/api/SearchResource.java

```diff
@@ -1,7 +1,22 @@
 package com.google.univiz.api;
 
+import java.util.List;
+
 /**
  * The SearchResoure interface is responsible for all data management/processing specific for the
  * search feature of the UniViz webapp.
  */
-public interface SearchResource {}
+public interface SearchResource {
+  /**
+   * Given a potentially incomplete search query, returns a list of colleges that would match the
+   * query.
+   *
+   * @param query The search query as the user typed in. It should be a non-empty and a non-null
+   *     string.
+   * @return An ordered list of possible matching colleges. The list is ordered such that the best
+   *     match is at the beginning of the list. The list may be empty which indicates there are no
+   *     possible matches. The list is also not an exhaustive set of matches. It only includes few
+   *     top results.
```

Comment:
> It only includes few top results

This is a great point - do you think this should be something that a client can configure? If so, consider adding it as an argument to this method.

Reply:
Probably! Ok, sounds good :) is that something I should handle in the front-end/servlet or backend?